src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -323,43 +330,48 @@ public List<JobModel> enumerateJobs(
   private List<FlightState> getAccessibleFlights(
       int offset, int limit, FlightFilter filter, AuthenticatedUserRequest userReq)
       throws InterruptedException {
-    int start = 0;
-    int end = offset + limit;
+
+    String nextPageToken = null;
     HashMap<String, List<String>> roleMap = new HashMap<>();
     ArrayList<FlightState> flightStateList = new ArrayList<>();
-    while (flightStateList.size() < offset + limit) {
-      List<FlightState> filterResults = stairway.getFlights(start, end, filter);
-      if (filterResults.size() == 0) {
+    int numTimesQueried = 0;
+    while (flightStateList.size() < offset + limit && numTimesQueried < MAX_FLIGHT_SEARCH_QUERIES) {
+      logger.info("Getting flight batch {} with npt {}", numTimesQueried, nextPageToken);
+      FlightEnumeration filterResults =
+          stairway.getFlights(nextPageToken, FLIGHT_SEARCH_BATCH_SIZE, filter);
```

Comment:
Thank you so much for getting Stairway flight fetch to work with a `nextPageToken`!

This is OOS for this PR, and may be on your radar for future work anyway…

But it would be awesome if our `enumerateJobs` endpoint supported the specification of a `nextPageToken` in addition to or instead of `offset`.  Then, fetching later pages in the job history would be as efficient as fetching most recent pages, since you don't have to determine access for all recent jobs first.

Reply:
Totally.  I was thinking that we'll likely want to rev the job api to a) do this and b) include total counts in the result (there's some stairway work I'd like to do first)